beater/server.go

```diff
@@ -218,14 +194,72 @@ func newGRPCServer(
 		batchProcessor,
 	}
 
-	jaeger.RegisterGRPCServices(srv, logger, batchProcessor, agentcfgFetcher)
-	if err := otlp.RegisterGRPCServices(srv, batchProcessor); err != nil {
+	srv.services = jaeger.RegisterGRPCServices(srv.Server, logger, batchProcessor, agentcfgFetcher)
+	if err := otlp.RegisterGRPCServices(srv.Server, batchProcessor); err != nil {
 		return nil, err
 	}
 	return srv, nil
 }
 
-func (s server) run() error {
+func (s *server) configure(
+	logger *logp.Logger,
+	info beat.Info,
+	cfg *config.Config,
+	tracer *apm.Tracer,
+	reporter publish.Reporter,
+	sourcemapStore *sourcemap.Store,
+	batchProcessor model.BatchProcessor,
+) error {
+	s.agentcfgFetchReporter = agentcfg.NewReporter(agentcfg.NewFetcher(cfg), batchProcessor, 30*time.Second)
+	ratelimitStore, err := ratelimit.NewStore(
+		cfg.AgentAuth.Anonymous.RateLimit.IPLimit,
+		cfg.AgentAuth.Anonymous.RateLimit.EventLimit,
+		3, // burst multiplier
+	)
+	if err != nil {
+		return err
+	}
+	*s.cfg = *cfg
```

Comment:
This looks like it'll create data races. What's synchronising access to the config?